src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -24,10 +20,24 @@
 from middlewared.plugins.directoryservices import DSStatus, SSL
 from middlewared.plugins.idmap import DSType
 import middlewared.utils.osc as osc
+
 try:
     from samba.dcerpc.messaging import MSG_WINBIND_ONLINE
+    from samba.credentials import Credentials, MUST_USE_KERBEROS
+    from samba.net import Net
+    from samba.samba3 import param
+    from samba.dcerpc import nbt
+    from samba import (gensec, ntstatus, NTSTATUSError)
 except ImportError:
     MSG_WINBIND_ONLINE = 9
+    MUST_USE_KERBEROS = 2
+    param = None
+    Net = None
+    Credentials = None
+    nbt = None
+    gensec = None
+
+LP_CTX = param.get_context()
```

Comment:
if the ImportError happens on SCALE we need this to be inside the try statement...

If not we should get rid of try/except at this point